packages/zoe/src/contracts/newSwap/collectFees.js

```diff
@@ -0,0 +1,28 @@
+// @ts-check
+
+import { amountMath } from '@agoric/ertp';
+import { trade } from '../../contractSupport';
+
+export const makeMakeCollectFeesInvitation = (zcf, feeSeat, centralBrand) => {
+  const collectFees = seat => {
+    const allocation = feeSeat.getAmountAllocated('RUN', centralBrand);
+    trade(
+      zcf,
+      {
+        seat,
+        gains: { RUN: allocation },
+      },
+      {
+        seat: feeSeat,
+        gains: { RUN: amountMath.makeEmpty(centralBrand) },
```

Comment:
This would be required for `reallocate` but not for `trade`

Reply:
Trade insisted on non-empty `gains`.